src/web/src/components/AuthProvider.tsx

```diff
@@ -117,6 +118,9 @@ const AuthProvider = ({ children }: Props) => {
           throw new Error(`login state '${state}' doesn't match expected state '${authState}'`);
         }
 
+        const supabase = createClient(supabaseUrl, anonKey);
+        // Override the JWT on the public client
+        supabase.auth.setAuth(accessToken);
```

Comment:
If we do this here, do we need to pass this `supabase` client out to be used?  For example, should we return it via:

```tsx
    <AuthContext.Provider value={{ login, logout, register, user, token, supabase }}>
      {children}
    </AuthContext.Provider>
```

Below on line 143?

Reply:
checking

Reply:
So you're right, we need to pass the client to be used in the context. Having trouble passing it to the return jsx though. I'm guessing I'd have to a line here for the client?

https://github.com/Seneca-CDOT/telescope/blob/8adb1386e81b78c361f3e7c9d342bff1693a35e0/src/web/src/components/AuthProvider.tsx#L9-L15

Reply:
`client? : ...` I'm not sure what to put here, out of practice with typescript

Reply:
Exactly, we need to update this.  You need to import the `SupabaseClient` type, then use that to add `supabase` to our interface:

```ts
import { createClient } from '@supabase/supabase-js'
...
export interface AuthContextInterface { 
   login: (returnTo?: string) => void; 
   logout: () => void; 
   register: (token: string) => void; 
   user?: User; 
   token?: string; 
   supabase?: SupabaseClient;
} 
```

Then you should be able to add your `supabase` instance to the object you return.